internal/search/query/literal_parser.go

```diff
@@ -91,6 +91,18 @@ loop:
 			// We see a space and the pattern is unbalanced, so assume this
 			// this space is still part of the pattern.
 			result = append(result, r)
+		case r == '\\':
+			// Handle escape sequence.
+			if len(buf) > 0 && len(buf[advance:]) > 0 {
```

Comment:
is advanced guaranteed to be less than len? Either way this can be one check:

```suggestion
			if len(buf) > advance {
```

Reply:
👍